### PR TITLE
Feature/colors and popup

### PR DESCRIPTION
**Pull request by szydlovsky**

### CODE REVIEW

File: Supplementary/Popup/PopupViewController.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    
+    override func loadView() {
+        super.loadView()
+        view = mainView
+    }
+    
+    override func viewDidLoad() {
+        mainView.onButtonPress = { [weak self] in
+            self?.model.action?()
+            self?.dismiss(animated: true)
```
Comment (avalevich) [inline Supplementary/Popup/PopupViewController.swift]: make it on the main thread?

File: Supplementary/Popup/PopupViewController.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        
+        mainView.onClosePress = { [weak self] in
+            self?.dismiss(animated: true)
```
Comment (avalevich) [inline Supplementary/Popup/PopupViewController.swift]: make it on the main thread?

File: Supplementary/Popup/PopupViewController.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        mainView.configure(
+            message: model.message,
+            buttonTitle: model.buttonTitle
+        )
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        let touch = touches.first
+        if touch?.view != mainView.containerView {
+            dismiss(animated: true)
```
Comment (avalevich) [inline Supplementary/Popup/PopupViewController.swift]: make it on the main thread?

File: Supplementary/Popup/PopupView.swift [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            messageLabel.bottomAnchor.constraint(equalTo: button.topAnchor, constant: -3),
+            messageLabel.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: 30),
+            messageLabel.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -30),
+            
+            closeButton.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 12),
+            closeButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -12)
+        ])
+        
+        button.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(buttonPressed)))
+        closeButton.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(closePressed)))
```
Comment (avalevich) [inline Supplementary/Popup/PopupView.swift]: change addGestureRecognizer to addTarget

**Review by avalevich — APPROVED** [review]

**Review by ura-dora — APPROVED** [review]

**Pull request merged**